File: skiko/src/jvmMain/java/org/jetbrains/awthrl/Components/Window.java
package org.jetbrains.awthrl.Components;

import java.awt.Graphics;
import javax.swing.JFrame;
import org.jetbrains.awthrl.DriverApi.Engine;

public class Window extends JFrame implements Drawable 
{   
    @Override
    public void paint(Graphics g) {
        Engine.get().render(this);
    }

    public void draw() { }

    public native void redrawLayer();

    public native void updateLayer();

    public native void disposeLayer();

    public native float getContentScale();
}

File: skiko/src/jvmMain/kotlin/org/jetbrains/awthrl/Components/Window.kt
package org.jetbrains.awthrl.Components

import org.jetbrains.awthrl.DriverApi.Engine
import java.awt.Graphics
import javax.swing.JFrame


open class Window : JFrame(), Drawable {
    override fun paint(g: Graphics) {
        Engine.get().render(this)
    }

    open fun draw() {}
    external override fun redrawLayer()
    external override fun updateLayer()
    external override fun disposeLayer()
    override val contentScale: Float
        external get
}